Modules/_datetimemodule.c

```diff
@@ -6537,136 +6514,130 @@ PyInit__datetime(void)
     };
 
     for (size_t i = 0; i < Py_ARRAY_LENGTH(types); i++) {
-        if (PyModule_AddType(m, types[i]) < 0) {
-            return NULL;
+        if (PyModule_AddType(module, types[i]) < 0) {
+            return -1;
         }
     }
 
     if (PyType_Ready(&PyDateTime_IsoCalendarDateType) < 0) {
-        return NULL;
+        return -1;
     }
     Py_INCREF(&PyDateTime_IsoCalendarDateType);
 
-    /* timedelta values */
-    d = PyDateTime_DeltaType.tp_dict;
 
-    x = new_delta(0, 0, 1, 0);
-    if (x == NULL || PyDict_SetItemString(d, "resolution", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+    PyObject *x = NULL;
 
-    x = new_delta(-MAX_DELTA_DAYS, 0, 0, 0);
-    if (x == NULL || PyDict_SetItemString(d, "min", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+#define DATETIME_ADD_MACRO(x, c)                    \
```

Comment:
The "d" variable is reassigned 3 times which makes the macro a little bit surprising. Would you mind to add a "dict" parameter to this macro?

Also, usually, such macro get the name first, and then an expression for the value. In _ctypes, I wrote:

```
#define MOD_ADD(name, expr) \
    do { \
        PyObject *obj = (expr); \
        if (obj == NULL) { \
            return -1; \
        } \
        if (PyModule_AddObjectRef(mod, name, obj) < 0) { \
            Py_DECREF(obj); \
            return -1; \
        } \
        Py_DECREF(obj); \
    } while (0)
```

I like that the variable scope if the scope of the macro call. It helps to check the reference count.

For example, I expect something usage like:

```
DATETIME_ADD_MACRO(d, "resolution", new_delta(0, 0, 1, 0));
```